### PR TITLE
FY2024 report fixes

### PR DESCRIPTION
**Pull request by eanders**
## Description

Various fix-ups for FY2024 changes

## Type of change
- [x] Bug fix
- [x] New feature (adds functionality)
- [x] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Review by gigxz — APPROVED** [review]

File: app/models/grda_warehouse/report/base.rb [hunk 1]
```diff
@@ -153,8 +154,8 @@ def self.sh_columns
         she_t[:project_type].to_sql,
         she_t[:project_tracking_method].to_sql,
         o_t[:id].as('organization_id').to_sql,
-        she_t[:housing_status_at_entry].to_sql,
-        she_t[:housing_status_at_exit].to_sql,
+        e_t[:LivingSituation].to_sql,
+        ex_t[:HousingAssessment].to_sql,
```
Comment (awisema) [inline app/models/grda_warehouse/report/base.rb:158]: Just confirming that this should not be`Destination`. Seems odd that at entry would be 3.917, but at exit is W5?

File: app/models/grda_warehouse/service_history_enrollment.rb [hunk 2]
```diff
@@ -233,11 +233,17 @@ def self.service_types
...
-      arel_table[:housing_status_at_entry].eq(5).
-        or(arel_table[:housing_status_at_exit].eq(5)),
-    )
+    # pre FY2024 logic, remove after we confirm the new version works
+    # where(
+    #   arel_table[:housing_status_at_entry].eq(5).
+    #     or(arel_table[:housing_status_at_exit].eq(5)),
+    # )
+    left_outer_joins(enrollment: :exit).
+      where(e_t[:LivingSituation].eq(205).or(ex_t[:HousingAssessment].eq(205)))
```
Comment (awisema) [inline app/models/grda_warehouse/service_history_enrollment.rb]: Same here… there is no way 205 for W5.1?

File: app/models/grda_warehouse/tasks/service_history/enrollment.rb [hunk 3]
```diff
@@ -430,8 +430,8 @@ def default_day
         project_name: project.ProjectName,
         organization_id: project.OrganizationID,
         record_type: nil,
-        housing_status_at_entry: self.LivingSituation,
-        housing_status_at_exit: exit&.HousingAssessment,
+        housing_status_at_entry: self.LivingSituation, # NOTE: we should remove this and rely on the join
+        housing_status_at_exit: exit&.HousingAssessment, # NOTE: we should remove this and rely on the join
```
Comment (awisema) [inline app/models/grda_warehouse/tasks/service_history/enrollment.rb:434]: Just flagging

File: drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb [hunk 4]
```diff
@@ -75,8 +75,8 @@ def she_household_column
       project_id: p_t[:id],
       project_name: :project_name,
       household_id: she_household_column,
-      housing_status_at_entry: :housing_status_at_entry,
-      housing_status_at_exit: :housing_status_at_exit,
+      housing_status_at_entry: :housing_status_at_entry, # NOTE: use Enrollment.LivingSituation in the future
+      housing_status_at_exit: :housing_status_at_exit, # NOTE: use Exit.HousingAssessment in the future
```
Comment (awisema) [inline drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb:79]: ?

**Review by awisema — CHANGES_REQUESTED** [review]
A question about housing status at exit.

File: app/models/grda_warehouse/report/base.rb (hunk 1 above)
Comment (eanders) [reply app/models/grda_warehouse/report/base.rb:158]: That is my reading of the notes we had.  (And that's what we were doing.)

File: app/models/grda_warehouse/service_history_enrollment.rb (hunk 2 above)
Comment (eanders) [reply app/models/grda_warehouse/service_history_enrollment.rb]: Right, that should have been 5 I believe, thanks!

File: app/models/grda_warehouse/tasks/service_history/enrollment.rb (hunk 3 above)
Comment (eanders) [reply app/models/grda_warehouse/tasks/service_history/enrollment.rb:434]: Yeah, I didn't want to pull out the assignment yet, but I _believe_ they are no longer in use.

File: drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb (hunk 4 above)
Comment (eanders) [reply drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb:79]: This is to remind us when we write the 2023 SPM (and copy the code from here) that we shouldn't be using that value any more.

Comment (eanders) [reply drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb:79]: Hmm, the previous SPM specs say:
System-specific tag for identifying projects serving Category 3 homeless
Dependent on specific HMIS product.
No current HMIS Data Standards indicator.

So maybe we came up with that?

**Pull request merged**